packages/expo-background-task/android/src/main/java/expo/modules/backgroundtask/BackgroundTaskConsumer.kt

```diff
@@ -0,0 +1,116 @@
+package expo.modules.backgroundtask
+
+import android.content.Context
+import android.util.Log
+import com.facebook.react.common.build.ReactBuildConfig
+import expo.modules.core.interfaces.LifecycleEventListener
+import expo.modules.interfaces.taskManager.TaskConsumer
+import expo.modules.interfaces.taskManager.TaskConsumerInterface
+import expo.modules.interfaces.taskManager.TaskExecutionCallback
+import expo.modules.interfaces.taskManager.TaskInterface
+import expo.modules.interfaces.taskManager.TaskManagerUtilsInterface
+import expo.modules.interfaces.taskManager.TaskServiceProviderHelper
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+const val BACKGROUND_TASK_TYPE = "expo-background-task"
+const val DEFAULT_INTERVAL_MINUTES = 60L * 24L // Once every day
+private val TAG = BackgroundTaskConsumer::class.java.simpleName
+
+class BackgroundTaskConsumer(context: Context?, taskManagerUtils: TaskManagerUtilsInterface?) :
+  TaskConsumer(context, taskManagerUtils), TaskConsumerInterface, LifecycleEventListener {
+
+  private var task: TaskInterface? = null
+  private var isBackgrounded: Boolean = true
+  private val taskCoroutineScope = CoroutineScope(Dispatchers.Default)
```

Comment:
We probably need to destroy this scope in `onHostDestroy` (or somewhere else - I'm not sure if the consumer can outlive the rn host) 🤔